components/LocalePicker.php

```diff
@@ -179,4 +177,40 @@ protected function makeLocaleUrlFromPage($locale)
 
         return $localeUrl;
     }
+
+    /**
+     * Makes sure to add any existing query string to the redirect url.
+     *
+     * @param $pageUrl
+     * @param $locale
+     *
+     * @return string
+     */
+    protected function withPreservedQueryString($pageUrl, $locale)
+    {
+        /**
+         * @event translate.localePicker.translateQuery
+         * Enables manipulating the URL query parameters
+         *
+         * You will have access to the page object, the old and new locale and the URL query parameters.
+         *
+         * Example usage:
+         *
+         *     Event::listen('translate.localePicker.translateQuery', function($page, $params, $oldLocale, $newLocale) {
+         *        if ($page->baseFileName == 'your-page-filename') {
+         *             return YourModel::translateParams($params, $oldLocale, $newLocale);
+         *         }
+         *     });
+         *
+         */            
+        $page = $this->getPage();
+        $query = request()->query();
+
+        $translatedQuery = Event::fire('translate.localePicker.translateQuery', 
```

Comment:
Event documentation needs to occur immediately before the firing of the event itself

Reply:
This event should also be documented in the README

Reply:
I have added a paragraph to the README and moved the comment. :+1: